dbt_platform_helper/domain/versioning.py

```diff
@@ -155,3 +160,40 @@ def _resolve_required_version(
             raise PlatformHelperVersionNotFoundException
 
         return out
+
+
+class AWSVersioning:
+    def __init__(self, latest_version_provider: VersionProvider = None):
+        self.latest_version_provider = latest_version_provider or GithubLatestVersionProvider
+
+    def get_version_status(self) -> VersionStatus:
+        aws_version = None
+        try:
+            response = subprocess.run("aws --version", capture_output=True, shell=True)
+            matched = re.match(r"aws-cli/([0-9.]+)", response.stdout.decode("utf8"))
+            aws_version = SemanticVersion.from_string(matched.group(1))
+        except ValueError:
+            pass
+
+        return VersionStatus(
+            aws_version, self.latest_version_provider.get_semantic_version("aws/aws-cli", True)
+        )
+
+
+class CopilotVersioning:
+    def __init__(self, latest_version_provider: VersionProvider = None):
+        self.latest_version_provider = latest_version_provider or GithubLatestVersionProvider
+
+    def get_version_status(self) -> VersionStatus:
+        copilot_version = None
+
+        try:
+            response = subprocess.run("copilot --version", capture_output=True, shell=True)
+            [copilot_version] = re.findall(r"[0-9.]+", response.stdout.decode("utf8"))
```

Comment:
is there a reason copilot_version is in a list?

Reply:
I am not too bothered, just not too used this syntax. Maybe it makes it slightly harder to read.